### PR TITLE
Game: Grid-based chapter select screen

### PR DESCRIPTION
**Pull request by teekaytai**
### Description

Changed the game chapter select screen to a grid-based one to reduce scrolling required by players.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to test

Run the Source Academy game and view the chapter select screen.

### Checklist

- [x] I have tested this code

### CODE REVIEW

**Review by lhw-1 — COMMENTED** [review]
UI looks clean and transition is seamless!

I did find a minor issue - this is when there are no chapters currently published:

Screenshot from 2022-02-08 11-26-17

At this point, when you click on the left arrow:

Screenshot from 2022-02-08 11-26-26

The page number becomes "1/0", and this is probably due to an edge case for `scrollPrevPage` function at `ChapterSelect.ts`, specifically the +1 on line 192.

Apart from this minor issue, LGTM! :+1: Will leave further approval to Gokul :smile:

File: src/features/game/scenes/chapterSelect/ChapterSelectHelper.ts [hunk 1]
```diff
@@ -103,7 +105,9 @@ export function createChapter(
     onUp: async () => await callGameManagerOnTxtLoad(true, index, lastCheckpointPlayed),
     onHover: () => chapterContinueHover.setVisible(true),
     onOut: () => chapterContinueHover.setVisible(false)
-  }).setPosition(-chapConstants.button.xOffset, chapConstants.button.yOffset);
+  })
+    .setPosition(chapConstants.playButton.xOffset, chapConstants.playButton.yOffset)
+    .setScale(0.65);
```
Comment (gok99) [inline src/features/game/scenes/chapterSelect/ChapterSelectHelper.ts:110]: Perhaps the button scale could also be abstracted into constants

File: src/features/game/scenes/chapterSelect/ChapterSelectHelper.ts [hunk 2]
```diff
@@ -152,8 +156,19 @@ export function createChapter(
...
-export function getCoorByChapter(chapterNum: number) {
-  const x = screenCenter.x + chapConstants.imageDist * chapterNum;
-  const y = screenCenter.y;
+function getCoorByChapter(chapterNum: number) {
+  const page = Math.floor(chapterNum / chapConstants.grid.chapPerPage);
+  chapterNum %= chapConstants.grid.chapPerPage;
+  const row = Math.floor(chapterNum / chapConstants.grid.chapPerRow);
+  const col = chapterNum % chapConstants.grid.chapPerRow;
+
+  const centreCol = chapConstants.grid.chapPerRow / 2 - 0.5;
```
Comment (gok99) [inline src/features/game/scenes/chapterSelect/ChapterSelectHelper.ts:165]: The 0.5 could seem a tad cryptic. Maybe `(row - 1) / 2` would be slightly clearer

**Review by gok99 — CHANGES_REQUESTED** [review]
Thanks for working on this! LGTM except for some very minor nitpicks and Hyung Woon's comment

**Comment by teekaytai** [comment]
Ok, fixed all the issues mentioned

**Review by gok99 — APPROVED** [review]

**Pull request merged**